README.md

```diff
@@ -117,47 +120,171 @@ If you'd like to get a message when the button is pressed or released, call the
+(chip select), CLK (clock), DIN (data in), and DOUT (data out).  We'll look at 
+the DIN line first to determine what to send.  Then we'll look at the DOUT line 
+to determine what we will receive (DOUT label is covered by the "1 Start bit" 
+box.
+
+![MCP3008 Data Sheet figure 5.1](assets/images/mcp3008-figure-5-1.PNG)
+
+The DIN line describes what the request should look like.  It should have 1 
+start bit, 1 mode bit, and 3 input address bits to select a channel.  The 
+MCP3008 has 8 (0-7) channels to which you can connect a sensor.  These are 
+described in greater detail in table 5.2 on page 19.
+
+![MCP3008 Data Sheet table 5.2](assets/images/mcp3008-table-5-2.PNG)
+
+For the sample below, we want to send a single-ended mode request to the 
+channel 2 (CH2) sensor.  So we need a start bit `1`, a `1` bit for the mode, and 
+3 bits, `010`, to select the sensor.  By comparing the DIN and DOUT lines to 
+the CLK line, you can tell there is 1 bit between the response and request.  For 
+that bit and the remaining bits, the DIN doesn't care.  We will send `0`
+bits in these places for the request.
+
+Now we move down to the response in the DOUT line.  We can see it begins with 
+a null bit `0` and then there are 10 bits of data.  The response is going to 
+be a binary number with a minumum value of 0 and a maximum value of 1023.  1023
+ is the maximum binary number that can be represented using 10 bits.
+
+We mentioned receiving and sending happen at the same time.  So we will begin 
+reading bits as soon as we send the request.  We will also send bits covering 
+the entire request/response cycle.  So our protocol (i.e. API) contains 5 bits 
+in the request, 1 bit between the request and response, 1 null bit, and 10 bits 
+of data.  Our protocol contains a total of 17 bits.
+
+However, we have a problem.  Our Raspberry Pi, along with most microcontrollers, 
+can only send requests using bytes (8 bits.)  We are 1 bit over a 2 byte 
+message.  Luckily the data sheet provides a solution to this problem in 
+section 6.0 APPLICATIONS INFORMATION.  The section goes into greater detail, 
+but figure 6.1 tell us how to modify the request/response to resolve the issue.
+
+![MCP3008 Data Sheet figure 6.1](assets/images/mcp3008-figure-6-1.PNG)
+
+The MCU Transmitted Data row tells us to send 7 `0` bits before the start 
+bit.  Combined with our 17 bit request/response communication, this will give us a 
+total of 24 bits and 3 bytes.  We can see from the MCU Received Data row we'll 
+need to take those 7 bits into account when we receive the data, but we'll also 
+receive 24 bits or 3 bytes.
+
+Using the protocol from the data sheet here is how we will read the moisture 
+sensor.
 
     # Make sure that you've enabled or loaded the SPI driver or this will
     # fail.
     iex> alias ElixirALE.SPI
     iex> {:ok, pid} = SPI.start_link("spidev0.0")
     {:ok, #PID<0.124.0>}
 
-    # Read the potentiometer
+    # Read the moisture sensor
```

Comment:
I'd keep the word "potentiometer", since that's what's in the schematic and on the board that I use to test Elixir ALE. From the software's point of view, it works the same.